### PR TITLE
docs: Add docs for stream feature views and stream data sources

### PR DESCRIPTION
**Pull request by felixwang9817**
Signed-off-by: Felix Wang <wangfelix98@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: docs/reference/data-sources/kinesis.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Kinesis source
+
+**Warning**: This is an _experimental_ feature. It's intended for early testing and feedback, and could change without warnings in future releases.
+
+## Description
+
+Kinesis sources allow users to register Kinesis streams as a data source. Feast currently does not manage the ingestion of data from Kinesis streams; users must manage the ingestion themselves and write feature values to the online store through [FeatureStore.write_to_online_store](https://rtd.feast.dev/en/latest/index.html#feast.feature_store.FeatureStore.write_to_online_store).
```
Comment (achals) [inline docs/reference/data-sources/kinesis.md:7]: Should clarify what we mean when we say that feast doesn't manage the ingestion jobs (like we don't launch or monitor them)

File: docs/reference/data-sources/kinesis.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+Kinesis sources must have a batch source specified, since that's the source used when retrieving historical features. When using a Kinesis source as a stream source in the definition of a feature view, a batch source doesn't need to be specified in the definition explicitly.
```
Comment (achals) [inline docs/reference/data-sources/kinesis.md:9]: Doesn't need to be specified in the feature view definition i presume?

File: docs/reference/data-sources/kafka.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+driver_stats_stream_source = KafkaSource(
+    name="driver_stats_stream",
+    kafka_bootstrap_servers="localhost:9092",
+    topic="drivers",
+    timestamp_field="event_timestamp",
+    batch_source=driver_stats_batch_source,
+    message_format=JsonFormat(
+        schema_json="driver_id integer, event_timestamp timestamp, conv_rate double, acc_rate double, created timestamp"
```
[the remark this answers is not in the document]
Comment (kevjumba) [reply docs/reference/data-sources/kafka.md:45]: This is valid, yes, this is the correct format for spark json format

File: docs/reference/data-sources/kinesis.md (hunk 1 above)
Comment (felixwang9817) [reply docs/reference/data-sources/kinesis.md:7]: clarified

File: docs/reference/data-sources/kinesis.md (hunk 2 above)
Comment (felixwang9817) [reply docs/reference/data-sources/kinesis.md:9]: clarified

File: docs/reference/data-sources/kafka.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Kafka sources allow users to register Kafka streams as a data source. Feast currently does not manage the ingestion of data from Kafka streams; users must manage the ingestion themselves and write feature values to the online store through [FeatureStore.write_to_online_store](https://rtd.feast.dev/en/latest/index.html#feast.feature_store.FeatureStore.write_to_online_store).
+
+Kafka sources must have a batch source specified, since that's the source used when retrieving historical features. When using a Kafka source as a stream source in the definition of a feature view, a batch source doesn't need to be specified in the definition explicitly.
+
+## Stream sources
+Streaming data sources are important sources of feature values. A typical setup with streaming data looks like:
+
+1. Raw events come in (stream 1)
+2. Streaming transformations applied (e.g. generating features like `last_N_purchased_categories`) (stream 2)
+3. Write stream 2 values to an offline store as a historical log for training
```
[the remark this answers is not in the document]
Comment (felixwang9817) [reply docs/reference/data-sources/kafka.md:16]: yeah, clarified

File: docs/reference/data-sources/kafka.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+4. Write stream 2 values to an online store for low latency feature serving
```
[the remark this answers is not in the document]
Comment (felixwang9817) [reply docs/reference/data-sources/kafka.md:17]: how is this optional? to be clear I'm interpreting this section more as a "here's how teams usually use stream data sources", not as "here's what the Feast API supports"

File: docs/reference/data-sources/kafka.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+5. Periodically materialize feature values from the offline store into the online store for improved correctness
```
[the remark this answers is not in the document]
Comment (felixwang9817) [reply docs/reference/data-sources/kafka.md:18]: clarified

**Pull request merged**